### PR TITLE
Bind `self.calculate_fid` for the case when FID is not requested.

### PR DESCRIPTION
**Pull request by Adversarian**
**Important**: Fixes an issue that was introduced in #227 (by yours truly and due to negligence 😭 ) where the class attribute `self.calculate_fid` was never bound unless `calculate_fid` was set to `True` in the Trainer which causes a fatal error in the case when FID is not requested from the Trainer.

### CODE REVIEW

File: denoising_diffusion_pytorch/denoising_diffusion_pytorch.py [hunk 1]
```diff
@@ -912,6 +912,7 @@ def __init__(
 
         # FID-score computation
 
+        self.calculate_fid = False
```
Comment (lucidrains) [inline denoising_diffusion_pytorch/denoising_diffusion_pytorch.py:915]: lol, not a big deal

here you can just set it once like

```python
self.calculate_fid = calculate_fid
if calculate_fid:
```

**Comment by lucidrains** [comment]
want to bump the patch version too? (i can immediately release the fix)

File: denoising_diffusion_pytorch/denoising_diffusion_pytorch.py (hunk 1 above)
Comment (Adversarian) [reply denoising_diffusion_pytorch/denoising_diffusion_pytorch.py:915]: Oh right yeah that went over my head for some reason.

Comment (Adversarian) [reply denoising_diffusion_pytorch/denoising_diffusion_pytorch.py:915]: Oh right yeah that went over my head for some reason.

**Comment by Adversarian** [comment]
Done.

**Pull request merged**